### PR TITLE
Update Span and Tracer interfaces; update tests

### PR DESCRIPTION
**Pull request by XerxesDGreat**
- Update Span and Tracer interfaces to conform to functionality which is assumed for all types
- Add phpunit configuration file
- Update tests to run more smoothly as a collection.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -6,12 +6,13 @@
         "ruafozy/mersenne-twister": "^1.3"
     },
     "require-dev": {
-        "phpdocumentor/phpdocumentor": "^2.8.5"
+        "phpdocumentor/phpdocumentor": "^2.8.5",
+        "phpunit/phpunit": "~4.8.20"
```
Comment (XerxesDGreat) [inline composer.json:10]: Using this, it is more clear on how tests are to be run; using composer will install phpunit

File: lib/Client/NoOpSpan.php [hunk 2]
```diff
@@ -6,6 +6,8 @@
 
 class NoOpSpan implements \LightStepBase\Span {
     public function guid() { return ""; }
+    public function traceGUID() { return ""; }
```
Comment (XerxesDGreat) [inline lib/Client/NoOpSpan.php:9]: needs to match parent span in order to be used interchangeably with the ClientSpan

File: lib/api.php [hunk 3]
```diff
@@ -53,7 +53,7 @@ public function startSpan($operationName, $fields);
      * @param  string $format the OpenTracing constant for the format of $carrier
      * @param  mixed $carrier the carrier object; the type depends on the $format
      */
-    public function inject($span, $format, &$carrier);
+    public function inject(Span $span, $format, &$carrier);
```
Comment (XerxesDGreat) [inline lib/api.php:56]: Makes for better protections against bad coding

File: lib/Client/ClientTracer.php [hunk 4]
```diff
@@ -282,7 +284,7 @@ public function join($operationName, $format, $carrier) {
             break;
 
         case LIGHTSTEP_FORMAT_BINARY:
-            throw new Exception('FORMAT_BINARY not yet implemented');
+            throw new \Exception('FORMAT_BINARY not yet implemented');
```
Comment (XerxesDGreat) [inline lib/Client/ClientTracer.php:287]: if using namespaces, better practice to include the `\` for classes like `Exception`

File: lib/api.php [hunk 5]
```diff
@@ -185,6 +191,20 @@ public function log($fields);
...
+     * @return string
+     */
+    public function traceGUID();
+
+    /**
+     * Sets the GUID of the containing trace onto this span.
+     *
+     * @param string $traceGUID
+     */
+    public function setTraceGUID($traceGUID);
```
Comment (XerxesDGreat) [inline lib/api.php:205]: These all seemed to be used by `ClientTracer`, regardless of whether a `NoOpSpan` or a `ClientSpan` was provided. I added them to make the interface consistent.

File: phpunit.xml.dist [hunk 6]
```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (XerxesDGreat) [inline phpunit.xml.dist:1]: configuration for the unit test run.

File: test/BaseLightStepTest.php [hunk 7]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+/**
+ * Class BaseLightStepTest
+ * @author Josh Wickham
+ * @copyright &copy; 2016 Life360, Inc.
+ * @since 6/10/16
+ */
+abstract class BaseLightStepTest extends PHPUnit_Framework_TestCase
```
Comment (XerxesDGreat) [inline test/BaseLightStepTest.php:9]: abstract base test class to add the `peek` so it's not a global function which would require tests being run in a particular way.

File: lib/api.php [hunk 8]
```diff
@@ -95,6 +95,13 @@ public function guid();
      * no-ops when in disabled mode.
      */
     public function disable();
+
+    /**
+     * Sets a list of options onto this trace
+     *
+     * @param array $options
+     */
+    public function options($options);
```
Comment (bcronin) [inline lib/api.php:104]: My one hesitation is on this one: I'm not sure we want `options()` to be part of the public API. The reason being that there are some options that can _only_ be set at initialization time -- and can't be changed at an arbitrary time on an Tracer instance. I want to be sure what whatever we support is made clear and we can _continue_ to support those options over time. 

Changing the comment to `Currently for internal use only` might alleviate my concerns about what we're signing up to support :)

**Comment by bcronin** [comment]
This looks awesome, Josh. Thanks!

**Pull request merged**